src/prpy/util.py

```diff
@@ -1326,7 +1326,7 @@ def ComputeGeodesicUnitTiming(traj, env=None, alpha=1.0):
     return new_traj
 
 
-def CheckJointLimits(robot, q, deterministic=None):
+def CheckJointLimits(robot, q, hard_limits, deterministic=None):
```

Comment:
I like @cdellin's suggestion quite a bit. It makes a lot of sense to implement this as a `MetaPlanner` that temporarily extends the robot's joint limits. It's also quite a bit simpler to implement than this pull request, since it only need be implemented once - instead of once per planner.